src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -238,121 +240,156 @@ func (gc *GarbageCollector) mark(ctx job.Context) error {
 	return nil
 }
 
+const (
+	BatchSize = 5
+)
+
 func (gc *GarbageCollector) sweep(ctx job.Context) error {
 	gc.logger = ctx.GetLogger()
-	sweepSize := int64(0)
-	blobCnt := 0
-	mfCnt := 0
+
+	// the mutex used when computing blobCnt, mfCnt, sweepSize
+	var mu sync.Mutex
+	blobCnt, mfCnt, sweepSize := new(int64), new(int64), new(int64)
+	*blobCnt, *mfCnt, *sweepSize = 0, 0, 0
+	runner1 := utils.NewLimitedConcurrentRunner(BatchSize)
```

Comment:
If we could use more semantic name to represent `runner1`, `runner2` like `blobRunner/artifactRunner`, that will better for understanding when go through code. But if it's hard to find a fit name, keep current name is 👌 .